### DISCUSSION TITLE
Use plain type for Zodios response instead of ZodType

### DISCUSSION THREAD

**Discussion by mphill** [ecyrbe/zodios#418] [Ideas]
The response is linked to a ZodSchema/ZodType, which seems unnecessary and also creates additional work.  If I am using an ORM it would be nice to use that type for the model as the response for Zodios instead of duplicating that ORM type into a ZodType. 

If the model/endpoint gets a user, you could assign a type of `Omit<User, "password">`.  It would cut down on a lot of boilerplate code! **I think for a majority of use cases Zod really only needs to be used for data coming into the API, not data being returned.**   If your database is trusted, running additional validation like ensuring an email is an email seems redundant.

Currently, Zod does not allow creating a Zod schema from a type, which could be used as a way to address this.

https://github.com/colinhacks/zod/issues/53

Essentially I want to use my ORM or SQL results as types to drive the type safety of the API response instead of Zod. Is this possible?

Thanks!

**Comment by ecyrbe** [comment]
Hello, No it's not possible.
Zodios is using contract based API definition, not Server first inferrence definitions.
Look at Zodios as OpenAPI for typescript. So type flow from contract (API definition) to Server and to Client.
This allows zodios to implement clients independantly from the server and server independantly from the client.
  
If you need types to flow from Server to Client (meaning, implementing server is mandatory), you have tRPC for that or if you want REST APIs there is hono

**Reply by ecyrbe** [reply]
But if you just want to type TS types instead of ZOD, take a look at zodios v11, we have pure TS schemas since we'll support any validation types even typescript : https://github.com/ecyrbe/zodios/blob/feature/v11/packages/core/examples/typescript-types.ts

**Reply by mphill** [reply]
I did consider tRPC since the return types do flow from server to client directly.  Zodios is so much easier to get set up and organized.  I like the fact you get the REST API included too.

https://github.com/colinhacks/tozod might be an option, but I will look at hono too. Thanks.

**Reply by mphill** [reply]
@ecyrbe this is exactly what I was looking for.  Thanks! I can work around some of these other things.  Really enjoying Zodios.  Great work and amazing DX.

**Reply by ecyrbe** [reply]
Thanks, becareful though v11 is still beta and documentation is just not yet ready. So you may encounter some difficulties with v11.
